library/src/scala/quoted/ValueOfExpr.scala

```diff
@@ -30,4 +32,252 @@ object ValueOfExpr {
     def apply(x: Expr[T])(given QuoteContext): Option[T] = matching.Const.unapply(x)
   }
 
+  given Option_delegate[T](given Type[T], ValueOfExpr[T]): ValueOfExpr[Option[T]] = new {
+    def apply(x: Expr[Option[T]])(given QuoteContext): Option[Option[T]] = x match {
+      case '{ None: Option[T] } => Some(None) // FIXME: remove ascription, Matcher should be able to match this expression
+      case '{ new Some[T](${Value(y)}) } => Some(Some(y))
+      case '{     Some[T](${Value(y)}) } => Some(Some(y))
+      case _ => None
+    }
+    override def toString(): String = "scala.quoted.ValueOfExpr.Option_delegate"
+  }
```

Comment:
Need to rename `*_delegate` , maybe in another PR.

Reply:
Agreed